src/components/Dashboard.test.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { render, cleanup, fireEvent } from "react-testing-library";
+
+import Dashboard from "./Dashboard";
+
+describe("The Dashboard component", () => {
+  it("should display four buttons: strike, ball, foul, hit", () => {
+    const { getByText } = render(<Dashboard />);
+    const buttons = ["Strike", "Ball", "Foul", "Hit"];
+    buttons.forEach(button => getByText(button));
+    cleanup();
+  });
+  describe("the strike button", () => {
+    it("should add 1 to strike when Strike button is pressed", () => {
+      const { getByText } = render(<Dashboard />);
+      const strikeButton = getByText("Strike");
+      fireEvent.click(strikeButton);
+      getByText("1");
+      cleanup();
+    });
+    it("should reset balls and strikes when strikes reach 3", () => {
+      const { getByText } = render(<Dashboard />);
+      const strikeButton = getByText("Strike");
+      fireEvent.click(strikeButton);
+      fireEvent.click(strikeButton);
+      fireEvent.click(strikeButton);
+      fireEvent.click(strikeButton);
+      getByText("1");
+      cleanup();
+    });
+  });
+  describe("the ball button", () => {
+    it("should add 1 to the ball count on each press", () => {
+      const { getByText } = render(<Dashboard />);
+      const ballButton = getByText("Ball");
+      fireEvent.click(ballButton);
+      fireEvent.click(ballButton);
+      getByText("2");
+      cleanup();
+    });
+    it("should reset count when 4 balls is reached", () => {
+      const { getByText } = render(<Dashboard />);
+      const ballButton = getByText("Ball");
+      const strikeButton = getByText("Strike");
+      fireEvent.click(strikeButton);
+      getByText("1");
+      for (let i = 1; i <= 6; i++) {
+        fireEvent.click(ballButton);
+      }
+      getByText("2");
+      getByText("0");
+      cleanup();
+    });
+  });
+  describe("the hit button", () => {
+    it("should reset the count when pressed", () => {
+      const { getByText, getByTestId } = render(<Dashboard />);
```

Comment:
I think you could have this in a `beforeEach`, [docs here](https://jestjs.io/docs/en/setup-teardown#repeating-setup-for-many-tests):
```javascript
let getByText, getByTestId;
beforeEach(() => {
  { getByText, getByTestId } = render(<Dashboard />);
});
```
You could also take advantage of `afterEach` for your cleanup:
```javascript
afterEach(cleanup);
```